src/main/java/com/fzakaria/mvn2nix/cmd/Maven2nix.java

```diff
@@ -39,71 +37,60 @@
     LoggingMixin loggingMixin;
 
     @Parameters(index = "0", paramLabel = "FILE", description = "The pom file to traverse.", defaultValue = "pom.xml")
-    private File file;
+    private File file = null;
+
+    @Option(names = "-goals", description = "The goals to execute for maven to collect dependencies.", defaultValue = "package")
+    private String[] goals;
+
+    @Option(names = "-repositories", description = "The maven repositories to try fetching artifacts from.", defaultValue = "https://repo.maven.apache.org/maven2/")
+    private String[] repositories;
```

Comment:
Rather than reading the settings file I just ask for the input from the CLI.
This punts on the challenge to the user but is more more configurable.

TODO: Figure out support for authenticated remote repositories